src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs

```diff
@@ -103,21 +98,21 @@ private static int GetNestedProxiesCount(HttpContext context, IFunctionExecution
                 return new NotFoundResult();
             }
 
-            if (context.Request.IsColdStart())
+            if (context.Request.IsColdStart() && !context.Items.ContainsKey(ScriptConstants.AzureFunctionsColdStartKey))
```

Comment:
This single line addresses the issue, however I also fixed up all the other Items.Add calls since to account for reentrancy.

I think there is potentially a deeper issue here in the way that this middleware is reentrant in the case of proxies. As you can see, none of the code modifying the context items bag is coded for reentrancy.

Reply:
Yes, that is problematic and was a concern with the original implementation. This needs to be revisited.